modules/500-upmeter/images/upmeter/pkg/agent/sender/client.go

```diff
@@ -89,11 +90,14 @@ func (c *Client) Send(reqBody []byte) error {
 	return nil
 }
 
-func NewHttpClient(config *ClientConfig, timeout time.Duration) *http.Client {
-	client, err := createSecureHttpClient(config.TLS, config.CAPath, timeout)
+func NewHttpClient(config *ClientConfig) *http.Client {
+	client, err := createSecureHttpClient(config.TLS, config.CAPath, config.Timeout)
 	if err != nil {
 		log.Errorf("falling back to default HTTP client: %v", err)
-		return &http.Client{Timeout: timeout}
+		return &http.Client{
+			Timeout:   config.Timeout,
+			Transport: newTransport(config.Timeout / 2),
```

Comment:
1. Why do we set timeout / 2 to read headers? For example, if there are only headers in the response, it's ok to spend the whole time reading headers.

2. Shouldn't other timeouts be set, e.g., connection timeout, handshake timeout?

3. There is the same division (timeout / 2) in two places. Maybe it is better to put this logic into the `newTransport` function? It makes more sense to me cos we will accept some generic timeout and then describe the logic of why we use `timeout / 2` for headers inside the function.

Reply:
1. Agreed. I removed the division.
2. I agree. That's why there was a copy of http.DefaultClient, where all the timeouts are set. And I changed only one that matters. It is done differently now, anyway.
3. I'd rather keep it explicit in two places than changing passed value implicitly under the hood. You pass a value, but it is halved! Anyway, I removed the division due to the first point.